packages/components/src/form-token-field/index.js

```diff
@@ -108,9 +108,14 @@ class FormTokenField extends Component {
 	}
 
 	onFocus( event ) {
+		const { expandOnFocus } = this.props;
 		// If focus is on the input or on the container, set the isActive state to true.
 		if ( this.input.hasFocus() || event.target === this.tokensAndInput ) {
-			this.setState( { isActive: true } );
+			const newState = { isActive: true };
+			if ( expandOnFocus ) {
+				newState.isExpanded = true;
+			}
+			this.setState( newState );
```

Comment:
Can this be done without mutating the object? Something like this?

```suggestion
			this.setState( {
				isActive: true,
				isExpanded: !! expandOnFocus || this.state.isExpanded,
			} );
```

Reply:
Good idea, changed in 71e13ed8d0549550a993d4f6bf77aea7891808d2.